ecosystem/typescript/sdk/src/utils/memoize-decorator.ts

```diff
@@ -77,22 +77,25 @@ function getNewFunction(
   return function (...args: any[]) {
     let returnedValue: any;
 
+    // @ts-ignore
+    const that: any = this;
```

Comment:
typescript has an explicit construct for typing `this`: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions . Prefer using this construct instead of the `that` aliasing

Reply:
memoize-decorator is borrowed code. It has so many anti-patterns. Previously, had a discussion with @banool, we are only going to do light touch on this library.